src/code/unix.lisp

```diff
@@ -1070,12 +1075,29 @@ the UNIX epoch (January 1st 1970.)"
                   current)))))))
 
   (declaim (inline system-internal-run-time))
-  #-sunos ; defined in sunos-os
+
+  ;; SunOS defines CLOCK_PROCESS_CPUTIME_ID but you get EINVAL if you try to use it,
+  ;; also use the same trick when clock_gettime should be avoided.
+  #-(or sunos avoid-clock-gettime)
```

Comment:
as a follow-up after this series, I think I'd suggest having the sunos build push the `avoid-clock-gettime` feature and simplifying this.